src/main/java/org/traccar/protocol/TotemProtocolEncoder.java

```diff
@@ -16,40 +16,27 @@
  */
 package org.traccar.protocol;
 
-import org.traccar.StringProtocolEncoder;
 import org.traccar.model.Command;
 import org.traccar.Protocol;
+import org.traccar.helper.Checksum;
 
-public class TotemProtocolEncoder extends StringProtocolEncoder {
+public class TotemProtocolEncoder extends TotemProtocolSmsEncoder {
 
     public TotemProtocolEncoder(Protocol protocol) {
         super(protocol);
     }
 
+    private String encodeCommand(String commandString) {
+        String builtCommand = String.format("$$%04dCF%s", 10 + commandString.getBytes().length, commandString);
+        return String.format("%s%02X", builtCommand, Checksum.xor(builtCommand));
+    }
+
     @Override
-    protected Object encodeCommand(Command command) {
+    protected String encodeCommand(Command command) {
```

Comment:
Please don't change the return type. It might break some refactoring later on if signature doesn't match.